frontend/packages/ux-editor-v3/src/components/config/editModal/EditBooleanValue.tsx

```diff
@@ -8,31 +8,36 @@ import { getComponentPropertyLabel } from '../../../utils/language';
 export interface EditBooleanValueProps extends IGenericEditComponent {
   propertyKey: string;
   helpText?: string;
+  defaultValue?: boolean;
 }
 
 export const EditBooleanValue = ({
   component,
   handleComponentChange,
   propertyKey,
   helpText,
+  defaultValue,
 }: EditBooleanValueProps) => {
   const t = useText();
 
   const handleChange = () => {
     handleComponentChange({
       ...component,
-      [propertyKey]: !component[propertyKey],
+      [propertyKey]: getNewBooleanValue(),
     });
   };
 
   const isValueExpression = (value: any) => {
     return Array.isArray(value);
   };
 
+  const getNewBooleanValue = () =>
+    component[propertyKey] === undefined ? !defaultValue : !component[propertyKey];
```

Comment:
Would it be easier to write it this way?
```suggestion
 const getNewBooleanValue = () => !(component[propertyKey] ?? defaultValue);
```